src/aap_eda/api/serializers/activation.py

```diff
@@ -604,10 +615,16 @@ def update(
             update_fields.append(key)
 
         instance.save(update_fields=update_fields)
-        if eda_credentials:
-            instance.eda_credentials.set(eda_credentials)
-        if event_streams:
-            instance.event_streams.set(event_streams)
+        if eda_credentials is not None:
```

Comment:
what's the difference between these 2 lines?
```
if eda_credentials is not None:
    if eda_credentials:
```
Same question for the check on `event_streams`

Reply:
There are three states. None means do nothing. Empty means clear existing value. A true list means replace with new value.